docs/core/tools/dotnet-aspnet-codegenerator.md

```diff
@@ -0,0 +1,191 @@
+```
+
+## Description
+
+The `dotnet aspnet-codegenerator ` global command runs the ASP.NET Core code generator and scaffolding engine.
+
+## Arguments
+
+`generator`
+
+The code generator to run. The following generators are available:
+
+| Generator | Operation |
+| ----------------- | ------------ | 
+| area      | [Scaffolds an Area](/aspnet/core/mvc/controllers/areas) |
+  controller| [Scaffolds a controller](/aspnet/core/tutorials/first-mvc-app/adding-model) |
+  identity  | [Scaffolds Identity](/aspnet/core/security/authentication/scaffold-identity) |
+  razorpage | [Scaffolds Razor Pages](/aspnet/core/tutorials/razor-pages/model) |
+  view      | [Scaffolds a view](/aspnet/core/mvc/views/overview) |
+
+## Options
+
+<!-- dotnet build to dotnet help use **bold** on options, most don't -->
+
+`-n|--nuget-package-dir`
+
+Specifies the NuGet package directory.
+
+`-c|--configuration {Debug|Release}`
+
+Defines the build configuration. The default value is `Debug`.
+
+`-tfm|--target-framework`
+
+Target [Framework](../../standard/frameworks.md) to use. For example, `net46`.
+
+<!-- REVIEW: Is this specified on the command line or in the project file? -->
+
+`-b|--build-base-path`
+
+The build base path.
+
+`-h|--help`
+
+Prints out a short help for the command.
+
+`--no-build`
+
+Doesn't build the project before running. It also implicit sets the `--no-restore` flag.
+
+`-p|--project <PATH>`
+
+Specifies the path of the project file to run (folder name or full path). If not specified, it defaults to the current directory.
+
+## Generator options
+
+The following sections detail the options available for the supported generators:
+
+* Area
+* Controller
+* Identity  
+* Razorpage
+* View
+
+<a name="area"></a>
+
+### Area options
+
+<!-- Review: I ran the following command in a Razor Pages project
+I was expecting Areas\AreaNameToGenerate\Pages
+Should I say this is intended for MVC projects and not RP projects
+-->
+Usage: `dotnet aspnet-codegenerator area AreaNameToGenerate`
+
+The preceding command generates the following folders:
+
+* *Areas*
+  * *AreaNameToGenerate*
+    * *Controllers*
+    * *Data*
+    * *Models*
+    * *Views*
+
+<a name="ctl"></a>
+
+### Controller options
+
+The following table lists options for  `aspnet-codegenerator` `controller` and `razorpage`:
+
+[!INCLUDE [aspnet-codegenerator-args-md.md](../../../includes/aspnet-codegenerator-args-md.md)]
+
+The following table lists options unique to  `aspnet-codegenerator controller`:
+
+| Option               | Description|
+| ----------------- | ------------ |
+| --controllerName or -name | Name of the controller. |
+| --useAsyncActions or -async | Generate async controller actions. |
+| --noViews or -nv | Generate **no** views. |
+| --restWithNoViews or -api  | Generate a Controller with REST style API. `noViews` is assumed and any view related options are ignored. |
+| --readWriteActions or -actions | Generate controller with read/write actions without a model. |
+
```

Comment:
--bootstrapVersion or -b is also available for all scaffolders except "Area". Its description is `Specify the bootstrap version. Valid values: '3', '4'. Default is 4.`

The bootstrap version will cause different content versions to be generated, appropriate for the specified bootstrap version. Also, if needed and not present, a wwwroot directory will be created that includes the specified bootstrap files.